docs/security/self-assessment.md

```diff
@@ -0,0 +1,167 @@
+3. **MetaManager:** the message processor between edged and edgehub. It is also responsible for storing/retrieving metadata to/from a lightweight database (SQLite). It contains MetaServer module, which realizes edge node to provide list&Watch capability externally.
+4. **DeviceTwin:** The module responsible for storing device status and syncing device status to the cloud. It also provides query interfaces for applications. It acts as a mirror to actual device metadata.
+5. **EventBus:** A MQTT client to interact with MQTT servers (mosquitto), offering publish and subscribe capabilities to other components.
+6. **ServiceBus:** ServiceBus is an HTTP client running on the edge. It receives requests from cloud services and interacts with the HTTP server running on the edge, enabling cloud services to access the HTTP server running on the edge over HTTP.
+
+## Self-assessment use
+
+This self-assessment is created by the KubeEdge team to perform an internal analysis of the project's security.  It is not intended to provide a security audit of KubeEdge, or function as an independent assessment or attestation of KubeEdge's security health.
+
+This document serves to provide KubeEdge users with an initial understanding of KubeEdge's security, where to find existing security documentation, KubeEdge plans for security, and general overview of KubeEdge security practices, both for development of KubeEdge as well as security of KubeEdge.
+
+This document provides the CNCF TAG-Security with an initial understanding of KubeEdge to assist in a joint-review, necessary for projects under incubation.  Taken together, this document and the joint-review serve as a cornerstone for if and when KubeEdge seeks graduation and is preparing for a security audit.
+
+## Security functions and features
+
+* Critical.  The cloud component CloudHub exposes ports to provide external services, accept connections from edge nodes, and be responsible for cloud-side communication. During communication, webSocket or Quick protocol is used for data transmission, and TLS certificate is used for data encryption.
+* Critical.  The EdgeHub component exposes the port to provide services externally, and provides list&Watch capability externally on the edge side.
+
+## Project compliance
+
+KubeEdge does not currently document meeting particular compliance standards.
+
+## Secure development practices
+
+KubeEdge has achieved the passing level criteria for [CII Best Practices](https://bestpractices.coreinfrastructure.org/en/projects/3018).
+
+### Development Pipeline
+
+All code is maintained in [Git](https://github.com/kubeedge/kubeedge) and changes must be reviewed by maintainers and must pass all unit and e2e tests. Code changes are submitted via Pull Requests (PRs) and must be signed. Commits to the `main` branch directly are not allowed.
+
+### Communication Channels
+
+* Internal. [GitHub Discussions](https://github.com/kubeedge/kubeedge/discussions)
```

Comment:
We currently don't use github discussions that much. Most communications happen through community meeting, slack, and github issues.

Reply:
Community meeting for example, we have regular meeting alternatve between Europe friendly time and Pacific friendly time. Besides, we have SIG meetings and on demond meetings. All are publicly accessable and meeting records are uploaded to youtube.